### DISCUSSION TITLE
v0.11.0

### DISCUSSION THREAD

**Discussion by AsamK** [AsamK/signal-cli#1015] [Releases]
<hr /><em>This discussion was created from the release <a href='https://github.com/AsamK/signal-cli/releases/tag/v0.11.0'>v0.11.0</a>.</em>

**Comment by stevie553** [comment]
First of all, thanks a lot for your great work on signal-cli :D

v0.11.0 broke building a fat jar: Execution failed for task ':fatJar'.
Entry META-INF/INDEX.LIST is a duplicate but no duplicate handling strategy has been set.

The "META-INF/INDEX.LIST" file is present in both "logback-classic-1.4.1.jar" and "logback-core-1.4.1.jar", I think this problem was introduced in https://github.com/AsamK/signal-cli/commit/2e8e81a926acd1e18829e280fbf5161cb0947ccc

I think maybe this is an upstream (logback) problem? I don't see any changes introduced in signal-cli other than bumping some dependencies that caused this. I could be wrong, I'm curious to hear your opinion, though.

**Reply by AsamK** [reply]
Yeah, seems to be caused by the logback update, though not really an upstream issue.
I've pushed a commit to master to ignore those files when building the fatJar. (They're just used for performance improvements when loading the jars)

**Reply by stevie553** [reply]
Thanks for the fix. The build passes. It may be worth noting that I got the following warnings:
Encountered duplicate path "META-INF/LICENSE.md" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "META-INF/NOTICE.md" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "META-INF/LICENSE.md" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "META-INF/NOTICE.md" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "META-INF/LICENSE.md" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "META-INF/NOTICE.md" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "META-INF/services/ch.qos.logback.classic.spi.Configurator" during copy operation configured with DuplicatesStrategy.WARN

I've added an * to the end of these file names https://github.com/AsamK/signal-cli/blob/333aacf4abc38f8f63e6fd92b4efb2d47be11234/build.gradle.kts#L77..L78 and that fixed the notice and license files warnings

Although the ch.qos.logback.classic.spi.Configurator warning is still there.

**Reply by AsamK** [reply]
That warning can be ignored for now. That file is required for the log configuration and it's currently using the correct one, not the logback default one.
It would be nice to exclude only the default logback file, but I haven't quite figured out how to do that yet.
